### ISSUE TITLE
Build doesn't happen on implicit global config (with no .yaml file present)

### ISSUE THREAD

**Issue by k-bx** [commercialhaskell/stack#687]
Just cloned hedis repo, and did:

```
➜  hedis git:(master) ✗ stack --version
Version 0.1.2.1, Git revision bf952bc6f6058f979c704680d34fa07a9f021778
➜  hedis git:(master) stack build
Run from outside a project, using implicit global config
Using resolver: lts-2.16 from global config file: /Users/kb/.stack/global/stack.yaml
```

Nothing happens after this message. This differs from what I would expect:

```
➜  hedis git:(master) stack init --resolver lts-2.16
Writing default config file to: /Users/kb/workspace/hedis/stack.yaml
Basing on cabal files:
- /Users/kb/workspace/hedis/hedis.cabal

Checking against build plan lts-2.16

* Build plan did not match your requirements:
    bytestring-lexing version 0.4.3.3 found
    - hedis requires >=0.5


Selected resolver: lts-2.16
Wrote project config to: /Users/kb/workspace/hedis/stack.yaml
➜  hedis git:(master) ✗ stack build
While constructing the BuildPlan the following exceptions were encountered:

--  Failure when adding dependencies:
      bytestring-lexing: needed (>=0.5), latest is 0.5.0.2, but 0.4.3.3 found
    needed for package: hedis-0.6.9
```

**Comment by manny-fp** [comment]
I'm thinking the correct behaviour would be emitting an error message telling the user to run `stack init`.  The only other behaviour I can think of that would make sense is to automatically do the equivalent of `stack init` before starting the build.  @snoyberg, opinion?

**Comment by snoyberg** [comment]
-1 on implicit stack init. Is the idea to raise an error whenever doing a
build with nothing to actually build?

**Comment by k-bx** [comment]
@snoyberg yeah, I think @manny-fp is proposing error-message that tells user needs to run `stack init`, no implicit stack init. Problem is that it's currently silently stops without any message.

**Comment by snoyberg** [comment]
I'm asking what the criteria is for the error message

**Comment by manny-fp** [comment]
I can think of a few different cases with different error messages:

1. In some random directory (like your home directory) - "specify a package on the command-line"
2. In a directory that looks like source code for a Cabal package (i.e., there's a `.cabal` file) - "run stack init" (wasn't there once a message along those lines?)
3. In a directory containing a stack.yaml with no listed `packages:` - "no packages in stack.yaml"

I'm not sure if there's any point trying to differentiate between (1) and (2) -- the error message can just say something like "specify a package on the command line, or run `stack init` if you're trying to build source in the current directory".

**Comment by snoyberg** [comment]
@k-bx Feel like taking a stab at this?

**Comment by k-bx** [comment]
Yeah, I'd be glad to try it out (starting Monday).

**Comment by k-bx** [comment]
Just wanted to update that I won't be able to work on this in next ~2 weeks, sorry.

**Comment by snoyberg** [comment]
See plan laid out at: https://github.com/commercialhaskell/stack/issues/651#issuecomment-129094820

**Comment by snoyberg** [comment]
Implemented for #651, please review PR #753.
